objcgen/EqualsHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+
+namespace ObjC {
+	public class EqualsHelper : MethodHelper{
+		public EqualsHelper (TextWriter headers, TextWriter implementation) : base (headers, implementation)
+		{
+			MonoSignature = "Equals (object)";
+			ReturnType = "bool";
+		}
+
+		public void WriteImplementation ()
+		{
+			BeginImplementation ();
+			implementation.WriteLine ("\tif (![other respondsToSelector: @selector (xamarinGetGCHandle)])");
```

Comment:
Already there, first test compares it to nil:

XCTAssertFalse ([c1 isEqual:nil], "equals nil");